engine/resources/mount.go

```diff
@@ -0,0 +1,712 @@
+
+	*obj = MountRes(raw) // restore from indirection with type conversion!
+	return nil
+}
+
+// strMapEq returns true, if and only if the two provided maps are identical.
+func strMapEq(x, y map[string]string) bool {
+	if len(x) != len(y) {
+		return false
+	}
+	for k, v := range x {
+		if val, ok := x[k]; !ok || v != val {
+			return false
+		}
+	}
+	return true
+}
+
+// fstabEntryExists checks whether or not a given mount exists in the provided
+// fstab file.
+func fstabEntryExists(file string, mount *fstab.Mount) (bool, error) {
+	mounts, err := fstab.ParseFile(file)
+	if err != nil {
+		return false, err
+	}
+	for _, m := range mounts {
+		if m.Equals(mount) {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+// fstabEntryAdd adds the given mount to the provided fstab file.
+func fstabEntryAdd(file string, mount *fstab.Mount) error {
+	mounts, err := fstab.ParseFile(file)
+	if err != nil {
+		errwrap.Wrapf(err, "error parsing file")
+	}
+	for _, m := range mounts {
+		// if the entry exists, we're done
+		if m.Equals(mount) {
+			return nil
+		}
+	}
+	// mount does not exist so we need to add it
+	mounts = append(mounts, mount)
+	return fstabWrite(file, mounts)
+}
+
+// fstabEntryRemove removes the given mount from the provided fstab file.
+func fstabEntryRemove(file string, mount *fstab.Mount) error {
+	mounts, err := fstab.ParseFile(file)
+	if err != nil {
+		errwrap.Wrapf(err, "error parsing file")
+	}
+	for i, m := range mounts {
+		// remove any entry with the defined mountpoint
+		if m.File == mount.File {
+			mounts = append(mounts[:i], mounts[i+1:]...)
+		}
+	}
+	return fstabWrite(file, mounts)
+}
+
+// fstabWrite generates an fstab file with the given mounts, and writes them
+// to the provided fstab file.
+func fstabWrite(file string, mounts fstab.Mounts) error {
+	// build the file contents
+	contents := fmt.Sprintf("# Generated by mgmt on ")
+	contents = contents + time.Now().Format(time.ANSIC) + "\n"
+	contents = contents + mounts.String() + "\n"
+	// write the file
+	if err := ioutil.WriteFile(file, []byte(contents), fstabUmask); err != nil {
+		return errwrap.Wrapf(err, "error writing fstab file")
+	}
+	return nil
+}
+
+// mountExists returns true, if a given mount exists in the given file
+// (typically /proc/mounts.)
+func mountExists(file string, mount *fstab.Mount) (bool, error) {
+	var err error
+	var m fstab.Mount
+	m = *mount // make a copy so we don't change the definition
+
+	// resolve the device's symlink if there is one
+	if m.Spec, err = evalSpec(mount.Spec); err != nil {
+		return false, errwrap.Wrapf(err, "error evaluating spec: %s", mount.Spec)
+	}
+
+	// get all mounts
+	mounts, err := fstab.ParseFile(file)
+	if err != nil {
+		return false, errwrap.Wrapf(err, "error parsing /proc/mounts")
+	}
+	// check for the defined mount
+	for _, p := range mounts {
+		found, err := mountCompare(&m, p)
+		if err != nil {
+			return false, errwrap.Wrapf(err, "error comparing mounts:\n %+v \n and \n %+v", mount, p)
```

Comment:
why \n ??

Reply:
It's easier to read, since its printing two structs. I can take them out if you prefer.

Reply:
yeah maybe something that prints nicer might be good. you might want to add a Debug + Logf if you want to print some longer thing like this.